### ISSUE TITLE
Context/right-click menu

### ISSUE THREAD

**Issue by GerardasB** [iTwin/iTwinUI#2157]
### Feature

AppUI team is deprecating `ContextMenu` from `@itwin/core-react` [storybook](https://itwin.github.io/appui/storybook/?path=/story/components-contextmenu--basic)
This component was regularly used to render an action list and can be usually replaced by an `iTwinUI` `DropdownMenu` i.e. saved views widget:
![Screenshot 2024-07-23 at 15 05 14](https://github.com/user-attachments/assets/fe7b19fc-6933-4ed0-a727-94777fbb5192)

Another use-case is when the menu is used as a contextual/right-click menu. Then it must be positioned without an explicitly visible target (i.e. following cursor), which `DropdownMenu` currently doesn't support out-of-the-box.
For this use case, we had a separate component [GlobalContextMenu](https://www.itwinjs.org/reference/core-react/contextmenu/globalcontextmenu/).

### Examples

Here's an example of how it could work with current iTwinUI components: [stackblitz](https://stackblitz.com/edit/github-fgpfaw-kdfj6e?file=src%2FApp.tsx).
Feature-wise the context menu is very similar to the `DropdownMenu` since it usually contains actions and can be nested.

**Comment by mayank99** [comment]
Hey @GerardasB, can you clarify the first use-case? The menu in your screenshot looks like a regular `DropdownMenu`, since it is triggered by a regular button.

**Comment by GerardasB** [comment]
> Hey @GerardasB, can you clarify the first use-case? The menu in your screenshot looks like a regular `DropdownMenu`, since it is triggered by a regular button.

Yes, that's just one of the use cases of how our current `ContextMenu` component was used and it should be replaced by iTwinUI `DropdownMenu`. We should focus on addressing the use case when there is no trigger.

**Comment by mayank99** [comment]
@MartynasStrazdas There is no update on it right now. We want to take the time to do it right, and we currently do not have the bandwidth for it.

However, what we can do in the meantime is expose a prop that allows changing the position of `DropdownMenu` so that you do not need to hack around it.

**Comment by jsnaras** [comment]
Hi, I am hereby adding an upvote to having a ContextMenu feature-equivalent component in iTwinUI. Thank you!

**Comment by mayank99** [comment]
[3.18.0](https://github.com/iTwin/iTwinUI/releases/tag/%40itwin%2Fitwinui-react%403.18.0) adds a new `positionReference` prop to `DropdownMenu`. This should avoid the need for any positioning hacks.

Note: The `DropdownMenu`'s primary purpose is still to serve as a regular (left-click) menu. If you are building a context (right click) menu, make sure that it's in addition to the primary menu.
